Stride-Labs/stride
# Inflation Fix

**Summary**
Fixes functionality after invariant checks


**Test plan**

Dockernet test - main functionality
1. Kick off RR changes by running `make start-docker build=s && make test-integration-docker`
2. Set max inner bound `$STRIDE_MAIN_CMD tx stakeibc set-redemption-rate-bounds GAIA 1 1.000001 --from $STRIDE_ADMIN_ACCT -y | TRIM_TX`
3. Observe zone is halted by querying it
```
➜  stride git:(inflation-fix) stridedl q stakeibc show-host-zone GAIA
host_zone:
  bech32prefix: cosmos
  chain_id: GAIA
...
  halted: true
```
4. Verify IBC transfer off of Stride fails
```
# try to ibc tokens off stride
$STRIDE_MAIN_CMD tx ibc-transfer transfer transfer channel-0 cosmos1wv34rdclakwnex6zarphdznmfgysa0s7ewr7va2xjx4ncd339eys34w877 1stuatom --from val1 -y 
# check tx log
raw_log: 'failed to execute message; message index: 0: denom stuatom is blacklisted:
```
6. Increase the max inner bound and unhalt the zone (1) `$STRIDE_MAIN_CMD tx stakeibc set-redemption-rate-bounds GAIA 1 1.300001 --from $STRIDE_ADMIN_ACCT -y | TRIM_TX` (2) `$STRIDE_MAIN_CMD tx stakeibc resume-host-zone GAIA --from $STRIDE_ADMIN_ACCT -y | TRIM_TX`
7. Verify IBC transfer off of Stride works using same IBC transfer as above and checking tx logs
```
tx:
  '@type': /cosmos.tx.v1beta1.Tx
  auth_info:
    fee:
      amount: []
      gas_limit: "200000"
      granter: ""
      payer: ""
    signer_infos:
    - mode_info:
        single:
          mode: SIGN_MODE_DIRECT
      public_key:
        '@type': /cosmos.crypto.secp256k1.PubKey
        key: ApSI1puwW2K3iA4+pk9UnBzAlpjglmbSNmoXMG/8P1vt
      sequence: "7"
    tip: null
  body:
    extension_options: []
    memo: ""
    messages:
    - '@type': /ibc.applications.transfer.v1.MsgTransfer
      memo: ""
      receiver: cosmos1wv34rdclakwnex6zarphdznmfgysa0s7ewr7va2xjx4ncd339eys34w877
      sender: stride1uk4ze0x4nvh4fk0xm4jdud58eqn4yxhrt52vv7
      source_channel: channel-0
      source_port: transfer
      timeout_height:
        revision_height: "1446"
        revision_number: "0"
      timeout_timestamp: "1697316508458020000"
      token:
        amount: "1"
        denom: stuatom
    non_critical_extension_options: []
    timeout_height: "0"
  signatures:
  - JUegOL6rcWL8Of3N49kGKlQphxllrHalORBCI3GU+SRC5ES/YDGrC17ws/RFPa3iLpj5QFrE+kKVsb18LAtyOw==
txhash: D884711E6C7011AB858B5046ED354A435F40DA2131225C503
```
8. Verify the host zone is not halted
```
➜  stride git:(inflation-fix) stridedl q stakeibc show-host-zone GAIA
host_zone:
  bech32prefix: cosmos
  chain_id: GAIA
...
  halted: false
```

Dockernet test - ledger
1. Remove admin check in `ValidateBasic`
2. Init a ledger account
3. Call the function `$STRIDE_MAIN_CMD tx stakeibc resume-host-zone GAIA --from ledger-test -y | TRIM_TX`
4. Check log
```
raw_log: 'failed to execute message; message index: 0: invalid chain id, zone for
  GAIA not halted: host zone is not halted'
```
(error indicates the signing happened correctly, which is what we care about when testing ledger)
